OpenRA.Mods.RA2/Traits/SpawnNeighboringActors.cs

```diff
@@ -0,0 +1,86 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA2.Traits
+{
+	[Desc("This actor places other actors around itself.")]
+	public class SpawnNeighboringActorsInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[ActorReference]
+		[Desc("Types of actors to place. If multiple are defined, a random one will be selected.")]
+		public readonly HashSet<string> ActorTypes = new HashSet<string>();
+
+		[FieldLoader.Require]
+		[Desc("Locations to place the mines, from top-left of the building.")]
+		public readonly CVec[] Locations = { };
+
+		public object Create(ActorInitializer init) { return new SpawnNeighboringActors(this, init.Self); }
+	}
+
+	public class SpawnNeighboringActors : INotifyKilled, INotifyOwnerChanged, INotifyActorDisposing
+	{
+		readonly SpawnNeighboringActorsInfo info;
+		readonly List<Actor> actors = new List<Actor>();
+
+		public SpawnNeighboringActors(SpawnNeighboringActorsInfo info, Actor self)
+		{
+			this.info = info;
+
+			SpawnActors(self);
+		}
+
+		public void SpawnActors(Actor self)
+		{
+			foreach (var offset in info.Locations)
+			{
+				var cell = self.Location + offset;
+				var actor = info.ActorTypes.Random(self.World.SharedRandom).ToLowerInvariant();
+				{
+					self.World.CreateActor(true, actor, new TypeDictionary
+					{
+						new OwnerInit(self.Owner),
+						new LocationInit(cell)
+					});
+				}
+			}
+		}
+
+		public void RemoveActors()
+		{
+			foreach (var actor in actors)
+				actor.Dispose();
+
+			actors.Clear();
+		}
+
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
```

Comment:
I don't see this documented at all and this seems to be a very specific behavior unrelated to spawning nearby actors.
Regardless of whether this functionality remains, it should be documented IMO.

Reply:
Added a note to the trait description.

Reply:
Is that in reference to the "which keep connected?"
That makes me think this'd be used for (visually) connecting walls, not ownership of the spawned actors always being that of the actor that spawned them throughout the game.

Reviewing this PR made me realize that the trait it introduces into RA2 seems like `FreeActor` with cell position offsets (instead of world) and selling, killing, and ownership changing of the free actors.
Would it make any sense to use (upgrading the implementation as necessary) multiple `FreeActor`s instead of a single similar `SpawnNeighboringActors`?

Reply:
Owner changing is still undocumented.